mcf-browser/src/back-end/parse.ts

```diff
@@ -128,41 +128,43 @@ async function getNodes(fileList: Blob[]) : Promise<ParseFileResponse> {
     } else if (fileExt === 'csv') {
       if (tmcfFile) {
         const tmcfFileName = (tmcfFile as File).name;
-        const tmcfOut = await ParseTmcf.generateMcf(tmcfFile, file).then(
-            (mcf) => {
-              const mcfParser = new ParseMcf(
-                  tmcfFileName + '&' + fileName,
-              );
-              return mcfParser.parseMcfStr(mcf as string);
-            },
-        );
 
-        const datapoints = await ParseTmcf.generateDataPoints(tmcfFile, file);
+        const parsedCsv = await ParseTmcf.parseTmcfAndCsv(tmcfFile, file);
+
+        const datapoints = parsedCsv.datapoints;
         finalReturn['datapoints'] = mergeDataPoints(
             finalReturn['datapoints'],
             datapoints,
         );
 
-        if (tmcfOut['errMsgs'].length !== 0) {
-          finalReturn['errMsgs'] = finalReturn['errMsgs'].concat({
-            file: tmcfFileName,
-            errs: tmcfOut['errMsgs'],
-          });
-        }
-        finalReturn['localNodes'] = finalReturn['localNodes'].concat(
-            tmcfOut['localNodes'],
+        const mcfs = parsedCsv.otherMcfs;
```

Comment:
defining this constant also feels unnecessary since it's only used in line 144 where you can just use `parsedCsv.otherMcfs`